js/noms/src/value.js

```diff
@@ -9,9 +9,10 @@ import type {primitive} from './primitives.js';
 import {ensureHash} from './get-hash.js';
 import type {Type} from './type.js';
 import type Ref from './ref.js';
-import type {WalkCallback} from './walk.js';
 import type {ValueReader} from './value-store.js';
 
+export type ValueCallback = (v: Value) => ?void | Promise<?void>;
```

Comment:
This type is strange. `?T` means `null | void | T` so `?void` is the same as `null | void | void` Do we want to allow `null` or not?